### PR TITLE
Upgraded com.hz.spi.impl.operationservice to Java 8

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl.java [hunk 1]
```diff
@@ -411,7 +411,7 @@ public boolean isCallTimedOut(Operation op) {
             Address owner = partitionService.getPartitionOwnerOrWait(partition);
 
             if (!memberPartitions.containsKey(owner)) {
-                memberPartitions.put(owner, new ArrayList<Integer>());
+                memberPartitions.put(owner, new ArrayList<>());
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl.java]: minor: the put-if-absent-then-add idiom can be substituted with more succinct `memberPartitions.computeIfAbsent(owner, k -> new ArrayList<>()).add(partition);`.

File: hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/Invocation_OffloadedTest.java [hunk 2]
```diff
@@ -100,21 +90,16 @@ public void start() {
     public void whenCompletesEventually() throws Exception {
         final String response = "someresponse";
 
-        InternalCompletableFuture<String> f = localOperationService.invokeOnPartition(new OffloadingOperation(new OffloadFactory() {
+        InternalCompletableFuture<String> f = localOperationService.invokeOnPartition(new OffloadingOperation(op -> new Offload(op) {
             @Override
-            public Offload create(Operation op) {
-                return new Offload(op) {
+            public void start() {
+                new Thread() {
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/Invocation_OffloadedTest.java]: minor: can be also converted to
```
                new Thread(() -> {
                    sleepSeconds(5);
                    offloadedOperation().sendResponse(response);
                }).start();
```

Comment (pveentjer) [reply hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/Invocation_OffloadedTest.java]: Converted

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl.java (hunk 1 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl.java]: Converted

**Pull request merged**